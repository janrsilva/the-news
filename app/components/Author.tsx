import React from "react";

const Author: React.FC<{ author: string; sourceName: string }> = ({
  author,
  sourceName,
}) => {
  return (
    <small className="text-right flex flex-col md:flex-row gap-1">
      <span>{author && " by " + author}</span>
      <span>{sourceName && " at " + sourceName}</span>
    </small>
  );
};

export default Author;
