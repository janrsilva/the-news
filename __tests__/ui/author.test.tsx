import { render } from "@testing-library/react";
import Author from "@/components/Author";

it("renders author and source name", () => {
  const { container } = render(
    <Author author="Jandsu Silva" sourceName="Uai News" />
  );
  expect(container).toMatchSnapshot();
});

it("renders author without source name", () => {
  const { container } = render(<Author author="Jandsu Silva" sourceName="" />);
  expect(container).toMatchSnapshot();
});
