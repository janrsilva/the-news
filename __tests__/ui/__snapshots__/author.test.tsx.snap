// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders author and source name 1`] = `
<div>
  <small>
     by Jandsu Silva
     at Uai News
  </small>
</div>
`;

exports[`renders author without source name 1`] = `
<div>
  <small>
     by Jandsu Silva
  </small>
</div>
`;
